src/main/java/blusunrize/immersiveengineering/common/blocks/wooden/TileEntityWindmill.java

```diff
@@ -100,6 +100,11 @@ protected float getSpeedModifier()
 
 	public boolean checkArea()
 	{
+		//Issue #2857 prevent crashing the server when EnumFacing.rotateY() would throw IllegalStateException.
+		//That happens when facing is UP or DOWN
+		if (EnumFacing.UP.equals(facing) || EnumFacing.DOWN.equals(facing))

```

Comment:
- Could be simplified to `facing.getAxis()==EnumFacing.Axis.Y`.
 - `equals` isn't necessary here as `EnumFacing` is an enum (as it says in the name)
 - The check should probably be at some earlier point as well as here. It should never be possible to place a windmill facing up or down.
 - We generally don't put comments on fairly obvious pieces of code (Ok, we don't put comments on non-obvious code either, but we should).